application/app/src/main/java/uk/nhs/digital/uec/api/authentication/localstub/LocalAmazonCognitoIdentityClientStub.java

```diff
@@ -0,0 +1,58 @@
+package uk.nhs.digital.uec.api.authentication.localstub;
+
+import static uk.nhs.digital.uec.api.authentication.constants.AuthenticationConstants.PASSWORD;
+import static uk.nhs.digital.uec.api.authentication.constants.AuthenticationConstants.USERNAME;
+import static uk.nhs.digital.uec.api.authentication.localstub.LocalConstants.COGNITO_GROUP;
+
+import com.amazonaws.services.cognitoidp.AbstractAWSCognitoIdentityProvider;
+import com.amazonaws.services.cognitoidp.model.AWSCognitoIdentityProviderException;
+import com.amazonaws.services.cognitoidp.model.AuthenticationResultType;
+import com.amazonaws.services.cognitoidp.model.InitiateAuthRequest;
+import com.amazonaws.services.cognitoidp.model.InitiateAuthResult;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class LocalAmazonCognitoIdentityClientStub extends AbstractAWSCognitoIdentityProvider {
+
+  private final Map<String, String> identityProviderIdPasswordMap;
+
+  public LocalAmazonCognitoIdentityClientStub() {
+    identityProviderIdPasswordMap = new HashMap<>();
+    identityProviderIdPasswordMap.put("admin@nhs.net", "password");
+  }
+
+  @Override
+  public InitiateAuthResult initiateAuth(InitiateAuthRequest initiateAuthRequest) {
+    Map<String, String> authParameters = initiateAuthRequest.getAuthParameters();
+    String inputUserName = authParameters.get(USERNAME);
+    String inputPassword = authParameters.get(PASSWORD);
+    String validPassword = identityProviderIdPasswordMap.get(inputUserName);
```

Comment:
Don't think this method works, as in the constructor you set only entry with the actual user name and password and in this method you use the keys "Username" and "Password"